src/Assignments/ProblemDetails.tsx

```diff
@@ -168,28 +168,49 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
                     logTag='gradedMessage'
                 >
                     {(currentMoment) => {
-                        if (_.isNil(grade) || _.isNil(problem)) {
-                            return (<></>);
-                        }
-
+                        // TODO move this logic to a utility function that is shared between the backend and front end
+                        // initally I was thinking the backend would send it, however it has to react to the current time so it probably would be better in a shared module
+                        const applicationError = 'An unknown error has occurred and it is unclear if your attempt will be graded.';
                         let message = null;
-                        if (grade.overallBestScore >= 1) {
+                        if (_.isNil(problem)) {
+                            // The user should never see this
+                            // TODO maybe problem shouldn't be able to be null?
+                            message = applicationError;
+                        }
+                        // ******************** NOT RECORDING ********************
+                        else if (_.isNil(grade)) {
+                            if (getUserRole() === UserRole.STUDENT) {
+                                message = 'There is no grade associated with this problem. Your attempts will not be recorded.';
```

Comment:
This seems a little strange to be a user-facing message, doesn't it? If you didn't know that a grade object had to exist in the database? I'm not sure what a better verbiage would be.